### PR TITLE
Fix #5983: Remove DEIS_* settings

### LINKED ISSUE #5983: Remove use of DEIS_* variables in settings
We can do the following instead:

1. Replace `DEIS_DOMAIN` with `CLUSTER_NAME` which is set in deployment.
2. Replace `DEIS_APP` with parsing the prefix of `HOSTNAME` which looks like e.g. `bedrock-stage-web-75c9dd854-pwc7x`

### PR DESCRIPTION
**Pull request by pmac**
Move to `CLUSTER_NAME` from the environment and `APP_NAME`
derived from `HOSTNAME`.

### CODE REVIEW

File: bedrock/settings/base.py [hunk 1]
```diff
@@ -366,11 +366,20 @@ def set_whitenoise_headers(headers, path, url):
     }
 }
 
+
+def get_app_name(hostname):
+    if hostname.startswith('bedrock-'):
```
Comment (jpetto) [inline bedrock/settings/base.py]: Maybe a comment here as to the naming convention/expected possible values? (You know, for the next time our infra changes :trollface:)

File: bedrock/settings/base.py [hunk 2]
```diff
@@ -366,11 +366,20 @@ def set_whitenoise_headers(headers, path, url):
...
+        return 'bedrock-' + app_mode
+
+    return None
+
+
 HOSTNAME = platform.node()
-DEIS_APP = config('DEIS_APP', default='')
-DEIS_DOMAIN = config('DEIS_DOMAIN', default='')
+APP_NAME = get_app_name(HOSTNAME)
+CLUSTER_NAME = config('CLUSTER_NAME', default='')
```
Comment (jpetto) [inline bedrock/settings/base.py]: Is there anything we need to do to ensure this is set on the clusters prior to merging/deploying this PR?

Comment (pmac) [reply bedrock/settings/base.py]: I took [this line in the configmap](https://github.com/mozmeao/www-config/blob/git-sync-operator/iowa-a/bedrock-prod/configmap.yaml#L10) to mean that it's ready. @jgmize can you confirm that it's ready and available in every cluster including oregon?

File: bedrock/settings/base.py [hunk 3]
```diff
@@ -366,11 +366,27 @@ def set_whitenoise_headers(headers, path, url):
     }
 }
 
+
+def get_app_name(hostname):
+    """
```
[the remark this answers is not in the document]
Comment (jpetto) [reply bedrock/settings/base.py]: Exactly! 🌟

File: bedrock/settings/base.py (hunk 2 above)
Comment (pmac) [reply bedrock/settings/base.py]: @jgmize actually I don't see `CLUSTER_NAME` in oregon deployments. I'll set that via deis commands and then merge.

**Pull request merged**